SPECS/gcc/gcc.spec

```diff
@@ -1,16 +1,74 @@
 %global security_hardening nofortify
 %define _use_internal_dependency_generator 0
+
+# Overriding the default to call 'configure' from subdirectories.
+%global _configure ../configure
+
+# Set if we're building cross-compilation packages for a given host architecture.
+%ifarch x86_64
+    %global build_cross 1
+%else
+    %global build_cross 0
+%endif
+
+# Adds a list of excluded files related to cross-compilation.
+# This macro is used only in the files list of the default 'gcc' package,
+# so that it doesn't include the cross-compilation files meant to go to
+# the 'gcc-<arch>' and 'gcc-c++-<arch>' subpackages (see: do_files() macro).
+#
+# Arguments:
+# - %1: name of the cross-compilation target architecture.
+# - %2: boolean indicating if we're building the cross-compilation bits for the current host architecture.
+#       See: "build_<architecture>" macros for each host architecture listed above.
+%global do_exclude() \
+%if %2 \
+%exclude %{_bindir}/%{1}* \
+%exclude %{_libdir}/gcc/%{1} \
+%exclude %{_libexecdir}/gcc/%{1} \
+%exclude %{_prefix}/%{1}/sys-root/ \
+%endif
+
+# Creates the files lists for the cross-compilation packages.
+#
+# Arguments:
+# - %1: name of the cross-compilation target architecture. This appears in the package and file names.
+# - %2: boolean indicating if we're building the cross-compilation bits for the current host architecture.
+#       See: "build_<architecture>" macros for each host architecture listed above.
+%global do_files() \
+%if %2 \
+%files -n gcc-%1 \
+%{_bindir}/%{1}*-cpp \
+%{_bindir}/%{1}*-gcc \
+%{_bindir}/%{1}*-gcc-ar \
+%{_bindir}/%{1}*-gcc-nm \
+%{_bindir}/%{1}*-gcc-ranlib \
+%{_bindir}/%{1}*-gcov* \
+%{_bindir}/%{1}*-lto-dump \
+%{_libdir}/gcc/%{1} \
+%{_libexecdir}/gcc/%{1} \
+%{_prefix}/%{1}/sys-root/ \
+\
+%files -n gcc-c++-%1 \
+%{_bindir}/%{1}*-c++ \
+%{_bindir}/%{1}*-g++ \
+%endif
+
 Summary:        Contains the GNU compiler collection
 Name:           gcc
 Version:        11.2.0
-Release:        7%{?dist}
+Release:        8%{?dist}
```

Comment:
Taking a look :)
[given its a big PR .. would take sometime .. hope it is fine]

Reply:
Sorry, can't really wait any longer and will merge now. If you see something worrying, let me know, and we can address it in a separate PR.